feat: Add dbt models for programs translations

SonaliBedge opened the pull request:
## Summary
Implements **[MFB-39](https://linear.app/myfriendben/issue/MFB-391/data-dbt-models-for-programs)1**: Create dbt models to reproduce the `data_programs.sql` materialized view output with row-level security by white_label_id.

This adds layered dbt models (`stg_programs_value_types`, `int_programs_value_types`, `mart_programs_value_types`) that extract program translation data (program value types) from Django's translation tables with white-label filtering support via RLS.

## Changes

### dbt Models (Layered Structure - Renamed for Clarity)
- **Staging**: `stg_programs_value_types.sql` - Extracts program value type translations with white_label_id from `translations_translation`, `translations_translation_translation`, and `programs_program` tables
- **Intermediate**: `int_programs_value_types.sql` - Passes through staging data with white_label_id preserved
- **Mart**: `mart_programs_value_types.sql` - Final table with RLS enabled, filtered to program value types only, restricted by white_label_id

### Configuration & Tests
- `mart_programs_value_types.yml` - Schema definition with data validation tests (not_null, unique, accepted_values, white_label_id constraint)
- `sources.yml` - Added `translations_translation`, `translations_translation_translation`, and `programs_program` source definitions


## Technical Notes
- **RLS Applied**: Program data is now white-label restricted via `setup_white_label_rls()` post-hook
- **White Label Support**: Added `white_label_id` column for multi-tenant access control
- **Materialization**: Mart model uses `table` materialization for performance
- **Join Logic**: Joins `programs_program` table to get white_label_id from value_type_id
- **Column Mapping**: Output columns (`id`, `master_id`, `label`, `language_code`, `text`, `white_label_id`) match original materialized view with added RLS support

## Testing
✅ All 9 tests pass (`dbt build --target postgres --select mart_programs_value_types+`)
✅ 150 rows created with white_label_id
✅ RLS functionality verified
✅ Model dependencies correctly configured

## Acceptance Criteria
- [x] Mart table created with RLS enabled by white_label_id
- [x] Models follow dbt layered structure (staging → intermediate → mart)
- [x] Tests included and passing (8 data tests on mart model)
- [x] White label filtering enabled for multi-tenant access control
- [x] File names clarified with `_value_types` suffix for specificity


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **Documentation**
  * Quick Start updated with explicit local setup steps: change into project, create/activate a Python virtual environment, install Python dependencies, and run dbt deps.

* **New Features**
  * Added program value-types pipeline: staging → intermediate → mart models to expose program value-type translations.
  * Multi-language translation support (filtered to en-us) and row-level security scoped by white_label.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

SonaliBedge committed:
feat: Add dbt models for programs translations

SonaliBedge committed:
Changed LEFT JOIN TO  an INNER JOIN in stg_programs.sql file

SonaliBedge committed:
Remove translation_type field. It is computed in int_programs.sql but not used in mart_programs.sql.

kbrey commented on dbt/models/postgres/staging/stg_programs.sql:
```diff
@@ -0,0 +1,33 @@
+    ttt.language_code,
+    ttt.text
+FROM translations_t tt
+INNER JOIN translations_tt ttt ON tt.translation_id = ttt.master_id
```
This matches the materialized view exactly! But after thinking through it some more, I think it would be really valuable to add white_label_id, so that we can add rls and be able to filter these program values by white label.

I think you can do that by adding a join in stg_programs to the programs_program table id which has white_label_id. Then you can pass white_label_id up to the mart and add the rls post_hook.

```
INNER JOIN {{ source('django_apps', 'programs_program') }} pp ON tt.translation_id = pp.value_type_id
```

kbrey commented on dbt/models/postgres/staging/stg_programs_value_types.sql:
This matches the materialized view name, but it took me awhile to figure out that this model returns just the value types. Maybe we could rename for clarity, like "stg_program_value_types" (similar for other models too), what do you think?

SonaliBedge replied:
Okay Katie, I will rename it to stg_program_value_types. It makes sense.

SonaliBedge replied:
@kbrey , I just want to confirm the naming changes:
stg_program → stg_program_value_types
mart_program → mart_program_value_types
int_programs → int_programs_value_types

kbrey reviewed (commented):
Nice work @SonaliBedge ! It works well and matches the existing view. I have a couple additional requests, let me know if you agree with them!

SonaliBedge committed:
Merge branch 'main' into sonali/mfb-391-data-dbt-models-for-programs

SonaliBedge committed:
Rename programs models to *_value_types for clarity

- Rename stg_programs.sql → stg_programs_value_types.sql
- Rename int_programs.sql → int_programs_value_types.sql
- Rename mart_programs.sql → mart_programs_value_types.sql
- Update all model references and YAML configurations
- Add white_label_id for RLS support

SonaliBedge committed:
Merge branch 'sonali/mfb-391-data-dbt-models-for-programs' of https://github.com/MyFriendBen/data-queries into sonali/mfb-391-data-dbt-models-for-programs

kbrey commented on dbt/models/postgres/marts/mart_programs_value_types.yml:
```diff
@@ -0,0 +1,34 @@
+        description: "Translation translation ID (primary key)"
+        tests:
+          - not_null
+          - unique
```
@SonaliBedge I'm getting a failed test for this, my guess is that there are multiple rows now that we are joining white label? Any ideas? It might be as easy as removing this test if it is no longer relevant.

SonaliBedge replied:
@kbrey , the test is not failing in my local environment. I just wanted to check whether you have any old data in the table, as the IDs are unique. Otherwise, I will remove this test.

kbrey replied:
@SonaliBedge apologies for the delay! I think you need to add the program_id column to these models, and check that for uniqueness instead of id. With the existing code, if two programs share a value type, you will fail the id uniqueness test. So in other words you need to (1) add program_id to all three models (2) move the unique test to the program_id column.

SonaliBedge replied:
Ok @kbrey . Thank you for reviewing—I’ll make the changes you suggested.

kbrey reviewed (commented):
This is great! One small issue with tests, otherwise I think it looks great.

SonaliBedge committed:
Made small changes

SonaliBedge committed:
feat(programs): add program_id as primary key to value_types models.

SonaliBedge committed:
Merge branch 'main' into sonali/mfb-391-data-dbt-models-for-programs

kbrey reviewed (approved):
Looks great, thanks for the changes!
